tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs

```diff
@@ -66,25 +67,23 @@ async Task OpenBugsForFailingBuilds(BugManagement bugManagement, IList<VstsBuild
         {
             // Filter out the builds for which we have already made bugs
             builds = FilterBuildsByDate(builds);
+            builds = FilterBuildsByStatus(builds);
             builds = FilterBuildsByExistingBugs(builds);
-            Console.WriteLine($"Filtering builds. {builds.Count} left for bug analysis after filtering");
+            Console.WriteLine($"Filtering builds complete. Creating bugs for {builds.Count} builds");
 
             // Create the bugs
             Dictionary<string, string> buildIdToBugId = new Dictionary<string, string>();
             foreach (VstsBuild build in builds)
             {
-                if (build.Result == VstsBuildResult.Failed && build.WasScheduled() == true)
```

Comment:
Moved this logic to a new function, as it is really just doing more filtering. It fits the pattern better to move it.